taccsite_cms/remote_cms_auth.py

```diff
@@ -0,0 +1,93 @@
+import inspect
+from django.http import HttpResponse, HttpResponseRedirect
+from django.contrib.auth.backends import ModelBackend
+from django.contrib.auth import get_user_model
+from django.contrib import auth as auth
+import requests
+import logging
+from django.conf import settings
+UserModel = get_user_model()
+
+logger = logging.getLogger(__name__)
+
+
+def verify_and_auth(request):
+    user = auth.authenticate(request)
+    if user:
+        # User is valid.  Set request.user and persist user in the session by logging the user in.
+        request.user = user
```

Comment:
is request.user not already user?

Reply:
At this point, the user has been authenticated but I don't think the user has been added to the request (unless the user already had a session, in which case it gets destroyed). Even the base ModelBackend implementation doesn't seem to add it to the request, I don't think. https://github.com/django/django/blob/2.2.16/django/contrib/auth/backends.py#L16